opm/simulators/wells/StandardWellEval.cpp

```diff
@@ -950,13 +951,26 @@ computeConnectionDensities(const std::vector<double>& perfComponentRates,
             if (!rvmax_perf.empty() && mix[gaspos] > 1e-12) {
                 rv = std::min(mix[oilpos]/mix[gaspos], rvmax_perf[perf]);
             }
-            if (rs != 0.0) {
+            double d = 1.0 - rs*rv;
+
+            if (d <= 0.0) {
+                std::ostringstream sstr;
+                sstr << "Problematic d value " << d << " obtained for well " << baseif_.name()
+                     << " during ccomputeConnectionDensities with rs " << rs
+                     << ", rv " << rv
+                     << " obtaining d " << d
+                     << " Continue as if no dissolution (rs = 0) and vaporization (rv = 0) "
+                     << " for this connection.";
+                deferred_logger.debug(sstr.str());
+            }
```

Comment:
Same with other places, and also the version in `MultisegmentWellEval.cpp`, `d > 0.0` can be removed by putting them in the `else` branch.